crates/astria-sequencer/src/bridge/bridge_lock_action.rs

```diff
@@ -95,26 +89,18 @@ impl ActionHandler for BridgeLockAction {
             .saturating_add(transfer_fee);
         ensure!(from_balance >= fee, "insufficient funds for fee payment");
 
-        // this performs the same checks as a normal `TransferAction`
-        transfer_check_stateful(&transfer_action, state, from).await
-    }
-
-    #[instrument(skip_all)]
-    async fn execute<S: StateWriteExt>(&self, state: &mut S, from: Address) -> Result<()> {
         let transfer_action = TransferAction {
             to: self.to,
             asset: self.asset.clone(),
             amount: self.amount,
             fee_asset: self.fee_asset.clone(),
         };
 
-        transfer_action
-            .execute(state, from)
-            .await
-            .context("failed to execute bridge lock action as transfer action")?;
+        check_transfer(&transfer_action, from, &state).await?;
+        execute_transfer(&transfer_action, from, &mut state).await?;
```

Comment:
There's a comment just below here saying "the transfer fee is already deducted in `transfer_action.execute()`" which should be updated to reflect this change.

Reply:
Thank you for noticing that. I left 2 `FIXME`s to separate out these logics.